packages/main/src/DatePicker.js

```diff
@@ -503,6 +503,102 @@ class DatePicker extends UI5Element {
 				this._toggleAndFocusInput();
 			}
 		}
+
+		if (this.isOpen()) {
+			return;
+		}
+
+		if (isPageUpShiftCtrl(event)) {
+			event.preventDefault();
+			this._changeDateValueWrapper(true, true, false, false);
+		} else if (isPageUpShift(event)) {
+			event.preventDefault();
+			this._changeDateValueWrapper(true, false, true, false);
+		} else if (isPageUp(event)) {
+			event.preventDefault();
+			this._changeDateValueWrapper(true, false, false, true);
+		}
+
+		if (isPageDownShiftCtrl(event)) {
+			event.preventDefault();
+			this._changeDateValueWrapper(false, true, false, false);
+		} else if (isPageDownShift(event)) {
+			event.preventDefault();
+			this._changeDateValueWrapper(false, false, true, false);
+		} else if (isPageDown(event)) {
+			event.preventDefault();
+			this._changeDateValueWrapper(false, false, false, true);
+		}
+	}
+
+	/**
+	 * Adds or extracts a given number of measuring units from the "dateValue" property value
+	 *
+	 * @param {boolean} years indicates that the measuring unit is in years
+	 * @param {boolean} months indicates that the measuring unit is in months
+	 * @param {boolean} days indicates that the measuring unit is in days
+	 * @param {boolean} forward if true indicates addition
+	 * @param {int} step number of measuring units to substract or add defaults to 1
+	 */
+	_changeDateValueWrapper(forward, years, months, days, step = 1) {
+		let date = this.dateValue;
+		date = this._changeDateValue(date, forward, years, months, days, step);
+		this.value = this.formatValue(date);
+		this.fireEvent("change", { value: this.value, valid: true });
+	}
+
+	/**
+	 * Adds or extracts a given number of measuring units from the "dateValue" property value
+	 *
+	 * @param {boolean} date js date object to be changed
+	 * @param {boolean} years indicates that the measuring unit is in years
+	 * @param {boolean} months indicates that the measuring unit is in months
+	 * @param {boolean} days indicates that the measuring unit is in days
+	 * @param {boolean} forward if true indicates addition
```

Comment:
forward is the 1st param, should be at the top

Reply:
Done